bin/truffleruby

```diff
@@ -0,0 +1,129 @@
+
+    if [ ${mx_time} -gt ${mvn_time} ]; then
+      exec mx -p "$root" ruby "$@"
+    fi
+fi
+
+if [ -z "$JAVACMD" ]
+then
+  if [ -z "$JAVA_HOME" ]
+  then
+    JAVACMD='java'
+  else
+    JAVACMD="$JAVA_HOME/bin/java"
+  fi
+fi
+
+# get the absolute path of the executable
+SELF_PATH=$(builtin cd -P -- "$(dirname -- "$0")" >/dev/null && pwd -P) && SELF_PATH=$SELF_PATH/$(basename -- "$0")
+
+# resolve symlinks
+while [ -h "$SELF_PATH" ]; do
+    # 1) cd to directory of the symlink
+    # 2) cd to the directory of where the symlink points
+    # 3) get the pwd
+    # 4) append the basename
+    DIR=$(dirname -- "$SELF_PATH")
+    SYM=$(readlink "$SELF_PATH")
+    SELF_PATH=$(cd "$DIR" && cd $(dirname -- "$SYM") && pwd)/$(basename -- "$SYM")
+done
+
+JRUBY_HOME_1=`dirname "$SELF_PATH"`           # the ./bin dir
+if [ "$JRUBY_HOME_1" = '.' ] ; then
+  cwd=`pwd`
+  JRUBY_HOME=`dirname $cwd` # JRUBY-2699
+else
+  JRUBY_HOME=`dirname "$JRUBY_HOME_1"`  # the . dir
+fi
+
+declare -a java_args
+declare -a ruby_args
+
+# Split out any -J argument for passing to the JVM.
+# Scanning for args is aborted by '--'.
+set -- $JRUBY_OPTS "$@"
+while [ $# -gt 0 ]
+do
+    case "$1" in
+    -J:*)
+        val=-${1:2}
+        if [ "${val}" = "-classpath" ]; then
+            CP="$CP:$2"
+            CLASSPATH=""
+            shift
+        elif [ "${val}" = "-cp" ]; then
+            CP="$CP:$2"
+            CLASSPATH=""
+            shift
+        else
+            if [ "${val:0:3}" = "-ea" ]; then
+                VERIFY_JRUBY="yes"
+            elif [ "${val:0:16}" = "-Dfile.encoding=" ]; then
+                JAVA_ENCODING=$val
+            fi
+            java_args=("${java_args[@]}" "${1:2}")
+        fi
+        ;;
+    -J*)
+        val=${1:2}
+        if [ "${val}" = "-classpath" ]; then
+            CP="$CP:$2"
+            CLASSPATH=""
+            shift
+        elif [ "${val}" = "-cp" ]; then
+            CP="$CP:$2"
+            CLASSPATH=""
+            shift
+        else
+            if [ "${val:0:3}" = "-ea" ]; then
+                VERIFY_JRUBY="yes"
+            elif [ "${val:0:16}" = "-Dfile.encoding=" ]; then
+                JAVA_ENCODING=$val
+            fi
+            java_args=("${java_args[@]}" "${1:2}")
+        fi
+        ;;
+     # Match switches that take an argument
+     -C|-e|-I|-S) ruby_args=("${ruby_args[@]}" "$1" "$2"); shift ;;
+     # Match same switches with argument stuck together
+     -e*|-I*|-S*) ruby_args=("${ruby_args[@]}" "$1" ) ;;
+     # Abort processing on the double dash
+     --) break ;;
+     # Other opts go to ruby
+     -*) ruby_args=("${ruby_args[@]}" "$1") ;;
+     # Abort processing on first non-opt arg
+     *) break ;;
+    esac
+    shift
+done
+
+# Append the rest of the arguments
+ruby_args=("${ruby_args[@]}" "$@")
+
+exec "$JAVACMD" $JAVA_OPTS "${java_args[@]}" "-Xbootclasspath/a:$JRUBY_HOME/lib/jruby-truffle.jar" org.truffleruby.Main "-Xhome=$JRUBY_HOME" "-Xlauncher=$%" "${ruby_args[@]}"
```

Comment:
This is getting simple, nice!
A bit of duplicated logic for `-J:` could probably be simplified later.